app/views/LoginSignupView.js

```diff
@@ -406,27 +409,39 @@ class LoginSignupView extends React.Component {
 	}
 
 	render() {
-		return (
-			<ScrollView style={[sharedStyles.containerScrollView, sharedStyles.container, styles.container]} {...scrollPersistTaps}>
-				<StatusBar />
-				<SafeAreaView testID='welcome-view' forceInset={{ vertical: 'never' }} style={styles.safeArea}>
-					{this.renderServices()}
-					{this.renderServicesSeparator()}
-					<Button
-						title={<Text>{I18n.t('Login_with')} <Text style={{ ...sharedStyles.textBold }}>{I18n.t('email')}</Text></Text>}
-						type='primary'
-						onPress={() => this.login()}
-						testID='welcome-view-login'
-					/>
-					<Button
-						title={I18n.t('Create_account')}
-						type='secondary'
-						onPress={() => this.register()}
-						testID='welcome-view-register'
-					/>
-				</SafeAreaView>
-			</ScrollView>
-		);
+		if (noEmailLogin === 1) {
+			return (
+				<ScrollView style={[sharedStyles.containerScrollView, sharedStyles.container, styles.container]} {...scrollPersistTaps}>
+					<StatusBar />
+					<SafeAreaView testID='welcome-view' forceInset={{ bottom: 'never' }} style={styles.safeArea}>
+						{this.renderServices()}
+						{this.renderServicesSeparator()}
+					</SafeAreaView>
+				</ScrollView>
+			);
+		} else {
+			return (
+				<ScrollView style={[sharedStyles.containerScrollView, sharedStyles.container, styles.container]} {...scrollPersistTaps}>
+					<StatusBar />
+					<SafeAreaView testID='welcome-view' forceInset={{ bottom: 'never' }} style={styles.safeArea}>
+						{this.renderServices()}
+						{this.renderServicesSeparator()}
+						<Button
+							title={<Text>{I18n.t('Login_with')} <Text style={{ ...sharedStyles.textBold }}>{I18n.t('email')}</Text></Text>}
+							type='primary'
+							onPress={() => this.login()}
+							testID='welcome-view-login'
+						/>
+						<Button
+							title={I18n.t('Create_account')}
+							type='secondary'
+							onPress={() => this.register()}
+							testID='welcome-view-register'
+						/>
```

Comment:
Is better if you wrap only buttons with `{ noEmailLogin ? renderButtons() : null }`, isn't needed duplicate `<SafeAreaView>` and others components.